src/base/bittorrent/session.cpp

```diff
@@ -1906,108 +1958,78 @@ bool Session::addTorrent_impl(CreateTorrentParams params, const MagnetUri &magne
-        // .torrent file to the .fastresume we loaded. Possibly from a
-        // failed upgrade.
-        return false;
     }
 
-    // We should not add torrent if it already
-    // processed or adding to session
-    if (m_addingTorrents.contains(hash) || m_loadedMetadata.contains(hash)) return false;
-
-    TorrentHandle *const torrent = m_torrents.value(hash);
-    if (torrent) {
-        if (torrent->isPrivate() || (!fromMagnetUri && torrentInfo.isPrivate()))
-            return false;
-        torrent->addTrackers(fromMagnetUri ? magnetUri.trackers() : torrentInfo.trackers());
-        torrent->addUrlSeeds(fromMagnetUri ? magnetUri.urlSeeds() : torrentInfo.urlSeeds());
-        return true;
-    }
-
-    qDebug("Adding torrent...");
-    qDebug(" -> Hash: %s", qUtf8Printable(hash));
-
-    // Preallocation mode
-    if (isPreallocationEnabled())
-        p.storage_mode = lt::storage_mode_allocate;
-    else
-        p.storage_mode = lt::storage_mode_sparse;
-
+    if (!hasCompleteFastresume) {
+        // Common
 #if (LIBTORRENT_VERSION_NUM < 10200)
-    p.flags |= lt::add_torrent_params::flag_paused; // Start in pause
-    p.flags &= ~lt::add_torrent_params::flag_auto_managed; // Because it is added in paused state
-    p.flags &= ~lt::add_torrent_params::flag_duplicate_is_error; // Already checked
+        p.flags |= lt::add_torrent_params::flag_paused; // Start in pause
+        p.flags &= ~lt::add_torrent_params::flag_auto_managed; // Because it is added in paused state
+        p.flags &= ~lt::add_torrent_params::flag_duplicate_is_error; // Already checked
 #else
-    p.flags |= lt::torrent_flags::paused; // Start in pause
-    p.flags &= ~lt::torrent_flags::auto_managed; // Because it is added in paused state
-    p.flags &= ~lt::torrent_flags::duplicate_is_error; // Already checked
+        p.flags |= lt::torrent_flags::paused; // Start in pause
+        p.flags &= ~lt::torrent_flags::auto_managed; // Because it is added in paused state
+        p.flags &= ~lt::torrent_flags::duplicate_is_error; // Already checked
 #endif
 
-    // Seeding mode
-    // Skip checking and directly start seeding (new in libtorrent v0.15)
-    if (params.skipChecking) {
-#if (LIBTORRENT_VERSION_NUM < 10200)
-        p.flags |= lt::add_torrent_params::flag_seed_mode;
-#else
-        p.flags |= lt::torrent_flags::seed_mode;
-#endif
-    }
-    else {
+        // Limits
+        p.max_connections = maxConnectionsPerTorrent();
+        p.max_uploads = maxUploadsPerTorrent();
+        p.save_path = Utils::Fs::toNativePath(savePath).toStdString();
+        p.upload_limit = params.uploadLimit;
+        p.download_limit = params.downloadLimit;
+
+        // Preallocation mode
+        p.storage_mode = isPreallocationEnabled()
+            ? lt::storage_mode_allocate : lt::storage_mode_sparse;
+
+        // Seeding mode
+        // Skip checking and directly start seeding (new in libtorrent v0.15)
```

Comment:
@Chocobo1:  **Relevancy Alert** 

`libtorrent v0.15` stopped being "new" back in the qBittorrent v2.x era (circa 2012).  

>Skip checking and directly start seeding **(new in libtorrent v0.15)** 

…should be… 

>Skip checking and directly start seeding